pdata/pcommon/map.go

```diff
@@ -22,20 +22,20 @@ import (
 )
 
 // Map stores a map of string keys to elements of Value type.
-type Map internal.Map
+type Map struct {
+	internal *internal.Map
```

Comment:
This does not work.. Since you will not be able in the `Span.Attributes` to return this type.

Reply:
We have `func NewMap(orig *[]otlpcommon.KeyValue, state *State, parent Parent[*[]otlpcommon.KeyValue]) Map` available in the internal package

Reply:
I see. We need a pointer to refresh the orig and state on any method. I reworked it and wrapped the pointer in a different way that doesn't change the existing interface, PTAL